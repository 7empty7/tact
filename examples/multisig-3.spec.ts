import { toNano } from "ton-core";
import { ContractSystem } from "ton-emulator";
import { MultisigContract } from "./output/multisig-3_MultisigContract";

describe('muiltisig-3', () => {
    it('should deploy', async () => {

        // Init contract
        let key1 = 1n;
        let key2 = 1n;
        let key3 = 1n;
        let system = await ContractSystem.create();
        let treasure = system.treasure('treasure');
        let contract = system.open(await MultisigContract.fromInit(key1, key2, key3));
        await contract.send(treasure, { value: toNano('10') }, 'Deploy');
        await system.run();

        // Check keys
        expect(await contract.getKey1()).toBe(key1);
        expect(await contract.getKey2()).toBe(key2);
        expect(await contract.getKey3()).toBe(key3);
    });
});