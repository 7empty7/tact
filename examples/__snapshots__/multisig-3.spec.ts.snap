// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`muiltisig-3 should deploy 1`] = `
[
  {
    "type": "deploy",
  },
  {
    "message": {
      "body": {
        "text": "Deploy",
        "type": "text",
      },
      "bounce": true,
      "from": "kQBbjpCWk51tkYZGsmjT-XQUL0QGlxJD4fQHGbYhdCbwQlX-",
      "to": "kQDuEv4CfVm1VY--rWckk55_MiS1rwXvE8k0rPx11-85US-l",
      "type": "internal",
      "value": 10000000000n,
    },
    "type": "received",
  },
  {
    "gasUsed": 4404n,
    "type": "processed",
  },
]
`;
